app/assets/javascripts/3dbio_viewer/src/data/repositories/BionotesPdbInfoRepository.ts

```
@@ -84,15 +102,63 @@ export class BionotesPdbInfoRepository implements PdbInfoRepository {
 
                 return buildPdbInfo({
                     id: pdbId,
-                    emdbs: emdbIds.map(emdbId => ({ id: emdbId })),
+                    emdbs: emdbs,
                     ligands: [],
-                    chains: [],
                     proteins,
-                    proteinsMapping: proteinsMappingChains,
+                    chainsMappings: proteinsMappingChains,
                 });
             });
         });
     }
+
+    private bionotesProteinMapping(
+        pdbId: string,
+        mapping: BioUniprotFromPdbMapping,
+        chains: ChainIds[]
+    ): MappingChain[] {
+        const proteins = mapping && mapping[pdbId.toLowerCase()];
+        if (!proteins || Array.isArray(proteins)) return chains;
+        else {
+            const proteinsMappingChains = _.map(proteins, (proteinChains, protein) =>
+                proteinChains.map(chainId => {
+                    const structAsymId = chains.find(c => c.chainId === chainId)?.structAsymId;
+                    if (!structAsymId) throw new Error("Missmatch between chains and proteins");
```

should we throw an error here? We are in a future method, this will simply blow the method with no recovery, is that what we want?

---

We should always have structAsymId defined. This error shouldn't happen and only will happen when there is something very critic external to us (ebi) that we aren't handling. This is one of the things that we should have in a test suite with all the PDB ids running each X time and if this happens we should address the error.

But, surely is not the best way to present the error to the user...